src/frontmatter.ts

```diff
@@ -0,0 +1,306 @@
+
+const formatSlug = (slug: string): string => slug.replace(/\.(mdx|md)$/, '');
+
+const isSupported = (
+  frontmatter: FrontMatter,
+  platformName: string,
+  guideName?: string
+): boolean => {
+  const canonical = guideName ? `${platformName}.${guideName}` : platformName;
+  if (frontmatter.supported && frontmatter.supported.length) {
+    if (frontmatter.supported.indexOf(canonical) !== -1) {
+      return true;
+    }
+    if (frontmatter.supported.indexOf(platformName) === -1) {
+      return false;
+    }
+  }
+  if (
+    frontmatter.notSupported &&
+    (frontmatter.notSupported.indexOf(canonical) !== -1 ||
+      frontmatter.notSupported.indexOf(platformName) !== -1)
+  ) {
+    return false;
+  }
+  return true;
+};
+
+let getDocsFrontMatterCache: Promise<FrontMatter[]> | undefined;
+
+export function getDocsFrontMatter(): Promise<FrontMatter[]> {
+  if (!getDocsFrontMatterCache) {
+    getDocsFrontMatterCache = getDocsFrontMatterUncached();
+  }
+  return getDocsFrontMatterCache;
+}
+
+async function getDocsFrontMatterUncached(): Promise<FrontMatter[]> {
+  const docsPath = path.join(root, 'docs');
+  const files = await getAllFilesRecursively(docsPath);
+  const allFrontMatter: FrontMatter[] = [];
+
+  // First, collect all non-common files
+  await Promise.all(
+    files.map(
+      limitFunction(
+        async file => {
+          const fileName = file.slice(docsPath.length + 1);
+          if (path.extname(fileName) !== '.md' && path.extname(fileName) !== '.mdx') {
+            return;
+          }
+
+          if (fileName.indexOf('/common/') !== -1) {
+            return;
+          }
+
+          const source = await readFile(file, 'utf8');
+          const {data: frontmatter} = matter(source);
+          allFrontMatter.push({
+            ...(frontmatter as FrontMatter),
+            slug: formatSlug(fileName),
+            sourcePath: path.join('docs', fileName),
+          });
+        },
+        {concurrency: FILE_CONCURRENCY_LIMIT}
+      )
+    )
+  );
+
+  // Add API documentation pages (categories and endpoints)
+  const categories = await apiCategories();
+  categories.forEach(category => {
+    allFrontMatter.push({
+      title: category.name,
+      slug: `api/${category.slug}`,
+    });
+
+    category.apis.forEach(api => {
+      allFrontMatter.push({
+        title: api.name,
+        slug: `api/${category.slug}/${api.slug}`,
+      });
+    });
+  });
+
+  // Now process all common files for each platform and expand them into platform/guide pages
+  const platformsPath = path.join(docsPath, 'platforms');
+  for await (const platform of await opendir(platformsPath)) {
+    if (platform.isFile()) {
+      continue;
+    }
+    const platformName = platform.name;
+
+    let platformFrontmatter: PlatformConfig = {};
+    const configPath = path.join(platformsPath, platformName, 'config.yml');
+    try {
+      platformFrontmatter = yaml.load(
+        await readFile(configPath, 'utf8')
```

Comment:
This is done in the loop. We can perhaps do this async via `Promise.allSettled` as well and batch all the `readFile` calls.

What maybe helps is precomputing the platforms via collecting the async iterator returned from `opendir`.

Reply:
fixed in [f253fa5](https://github.com/getsentry/sentry-docs/pull/15452/commits/f253fa56f1de18a8241fd96086b986ae8e9f1f1a)